---
title : "10.함수에서 에러가 발생하는 경우"
categories:
  - 파이썬(구조)
# tags:
#  - Python
#  - Python구조

date : 2020-11-29
---

파이썬에서 함수에서 에러가 발생하는 경우와 에러를 어떻게 처리하는 지에 대해 알아보자.  

Error 발생 예시 
--- 

에러1 인덱스 첨자 범위를 벗어나는 경우  

```python 
>>> a=[10,20,30]
>>> a[3]
Traceback (most recent call last):
  File "<pyshell#1>", line 1, in <module>
    a[3]
IndexError: list index out of range    #범위를 벗어나는 경우의 error 메세지 

```

에러2 형식이 잘못 된 경우  

```python 
>>> result=5/'string'
Traceback (most recent call last):
  File "<pyshell#2>", line 1, in <module>
    result=5/'string'
TypeError: unsupported operand type(s) for /: 'int' and 'str'  #형식이 잘못된 경우의 error 메세지 

```
에러3 0으로 값을 나누는 경우  

```python
>>> result=5/0
Traceback (most recent call last):
  File "<pyshell#10>", line 1, in <module>
    result=5/0
ZeroDivisionError: division by zero     #0으로 나누는 것을 시도했을 때 error message

```

Error 처리 방법
--- 

파이썬에서는 에러처리를 위한 클래스 계층이 제공된다.  아래와 같은 다양한 예외처리를 위한 에러클래스가 제공되고 해당 클래스를 통해 Error를 출력시켜준다.    
위에서 확인한 error message 마지막줄에 출력하고 있는 것도 Error class들 중 하나이다.  
(IndexError, TypeError... )  

```python 
>>> BaseException
<class 'BaseException'>
>>> SystemExit
<class 'SystemExit'>
>>> StopIteration
<class 'StopIteration'>
>>> ArithmeticError
<class 'ArithmeticError'>

```
